### ISSUE TITLE
Extreme Confidence Values in Voxel Predictions during Segmentation

### ISSUE THREAD

**Issue by chenney0830** [MIC-DKFZ/nnUNet#2763]
Hello,

I'm experiencing an issue while using nnU-Net for brain tumor segmentation. After training the model and performing inference, I used the --save_probability option to save the confidence predictions for each voxel. However, I noticed that the confidence values are extremely polarized—almost all voxels are predicted as either 0.0001 or 0.9999, with only the edges showing values around 0.8.

Could you please advise what might be causing this behavior and suggest any potential solutions?

Thank you for your help!

**Comment by sten2lu** [comment]
Hi @chenney0830,

The general reasons for this are the following:
1. nnU-Net inference computes the mean over the logits instead of over the probabilities (example below for binary case based on sigmoid)
2. nnU-Net uses both Cross-Entropy and DICE loss. The DICE-loss is known to decrease the calibration of a model
3. Neural Networks are generally known to be overconfident even when trained purely with Cross-Entropy


Depending on your task you could do the following:
A. Temperature scaling of the logits of the model in a post-hoc form on a validation set (requires you to access the logits).
B. Platt Scaling the Probabilities in a post-hoc (rounding during softmax might lead to issues here).
C. Train without the DICE-loss and purely use the Cross-Entropy loss.

Best regards,
Carsten
